map/src/menu/settings/CloudChanges.jsx

```diff
@@ -0,0 +1,169 @@
+import React, { useCallback, useContext, useEffect, useRef, useState } from 'react';
+import {
+    AppBar,
+    Box,
+    Divider,
+    IconButton,
+    ListItemIcon,
+    ListItemText,
+    MenuItem,
+    Skeleton,
+    Toolbar,
+    Typography,
+} from '@mui/material';
+import { useInView } from 'react-intersection-observer';
+import { ReactComponent as BackIcon } from '../../assets/icons/ic_arrow_back.svg';
+import { ReactComponent as MenuIcon } from '../../assets/icons/ic_overflow_menu_white.svg';
+import { ReactComponent as MenuIconHover } from '../../assets/icons/ic_overflow_menu_with_background.svg';
+import MenuItemsTitle from '../components/MenuItemsTitle';
+import AppContext from '../../context/AppContext';
+import headerStyles from '../trackfavmenu.module.css';
+import styles from './settings.module.css';
+import trackStyles from '../../menu/trackfavmenu.module.css';
+import { useTranslation } from 'react-i18next';
+import Loading from '../errors/Loading';
+import TracksManager from '../../manager/track/TracksManager';
+import ActionsMenu from '../actions/ActionsMenu';
+import ChangesActions from '../actions/ChangesActions';
+import { formatDate, getFileItemSize, getItemIcon } from '../../manager/SettingsManager';
+import { refreshGlobalFiles } from '../../manager/track/SaveTrackManager';
+import Empty from '../errors/Empty';
+
+export default function CloudChanges({ files, setOpenCloudSettings, filesLoading }) {
+    const ctx = useContext(AppContext);
+    const { t } = useTranslation();
+
+    const [changes, setChanges] = useState(files);
+
+    useEffect(() => {
+        setChanges(files);
+    }, [files]);
+
+    function closeChanges() {
+        setOpenCloudSettings(false);
+        refreshGlobalFiles({ ctx }).then();
+    }
+
+    const ChangesItem = React.memo(({ item }) => {
```

Comment:
2 cases of using React.memo. Possibly better to switch to useMemo() with non-object dependency. Let's discuss in PM.

Reply:
useMemo is a hook for memoizing values within a component to avoid expensive recalculations on re-renders with unchanged dependencies. React.memo is a higher-order component that prevents re-rendering if a component's props haven't changed, optimizing performance. Thus, I chose React.memo to optimize rendering behavior based on prop changes, not for memoizing internal calculations.

Reply:
Ok. We've never used React.memo for this purpose before.

Using useMemo was enough to create memoized child components.

Anyway React.memo might be OK, but please test how React.memo works in your case:

- item is not an object so possibly you can't use it directly as dependency (for example, you definitely can not use this item in useEffect as a ..., [dep])

- try logging and monitor how often both React.memo() components really rendered (use 1st console.log before React.memo and 2nd console.log just inside component's { })